cruft/_commands/_generate.py

```diff
@@ -0,0 +1,119 @@
+import json
```

Comment:
Can you please create a separate commit with the common parts that have been moved? It will make it easier to review. So the way I'd do it is this. Retractor out the common parts from update in a separate commit. Then once that is done, implement the diff command in a separate commit.

Reply:
Done.